modules/seqexec/server/src/main/scala/seqexec/server/SeqTranslate.scala

```diff
@@ -69,7 +69,7 @@ class SeqTranslate(site: Site, systems: Systems[IO], settings: TranslateSettings
       headers: HeaderExtraData => List[Header[IO]],
       stepType: StepType
     ): SequenceGen.StepGen[IO] = {
-      val ia = inst.instrumentActions(config)
+      val ia = inst.instrumentActions(config, sys.collectFirst{case x: Tcs[IO] => x})
```

Comment:
Doesn't this give you an erasure error?

Reply:
I was afraid it would, but it compiles.

Reply:
That's unsound, you need a better solution for that.

Reply:
Sorry, I misunderstood that code. It's actually ok, it's just weird. But Scala handles that case ok. You would get a warning for anything other than `IO` in that position.

Reply:
I recommended to parametrize the method anyway and move it to `ObserveEnvironment[F]`